lib/Etsy/StatsD.pm

```diff
@@ -114,19 +185,16 @@ sub send {
 
 	#failures in any of this can be silently ignored
 	my $count  = 0;
-	my $socket = $self->{socket};
-	while ( my ( $stat, $value ) = each %$sampled_data ) {
-		_send_to_sock($socket, "$stat:$value\n");
-		++$count;
-	}
+	foreach my $socket ( @{ $self->{sockets} } ) {
+        # Using foreach/keys because each iterator is unreliable when nested
```

Comment:
The problem is they are iterating the same collection, $sampled_data.  I'm modifying the code to use the iterator with a keys() void context call to ensure the iterator is reset.  According to the docs,  a void context keys() call just resets the iterator with zero overhead.